lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -121,13 +128,80 @@ class CombineSelectMaskedLoadPattern : public RewritePattern {
     if (!mask)
       return failure();
 
-    auto splatOp = mask.getDefiningOp<SplatOp>();
-    if (!splatOp)
-      return failure();
+    bool findTargetPattern = false;
+    if (mask == condSelect) {
+      // Case 1: The mask of the load and the cond of the select are the same
+      // value.
+      // ```mlir
+      //   %mask : i1
+      //   %true_val = tt.load %ptr, %mask, %other : !tt.ptr<f32>
+      //   %res = arith.select %mask, %true_val, %false_val : f32
+      // ```
+      findTargetPattern = true;
+    } else if (auto splatOp = mask.getDefiningOp<SplatOp>()) {
+      // Case 2: The mask of the load is splatted from the cond of the select.
+      // ```mlir
+      //   %mask = tt.splat %cond : i1 -> tensor<8xi1>
+      //   %true_val = tt.load %ptr, %mask : tensor<8x!tt.ptr<f32>>
+      //   %res = arith.select %cond, %true_val, %false_val : tensor<8xf32>
+      // ```
+      if (splatOp.getSrc() == condSelect) {
+        findTargetPattern = true;
+      }
+    }
 
-    auto splatCond = splatOp.getSrc();
-    if (splatCond != condSelect)
+    // Case 3: The condition of the select is a value extracted from a dense
+    // tensor (the mask value).
+    // ```mlir
+    //   %mask : dense tensor(splatted from same value)
+    //   %true_val = tt.load %ptr, %mask
+    //   %cond = tensor.extract %mask[...]
+    //   %res = arith.select %cond, %true_val, %false_val
+    // ```
+    if (auto extractOp = condSelect.getDefiningOp<tensor::ExtractOp>()) {
```

Comment:
triton doesn't supprt tensor.extract

Reply:
Does Triton have any plans to use `tensor.extract` in the future to get a specific value from a tensor?

I believe this could have many practical applications. 

For example, scalars could be treated as 0-rank tensors using `tensor.from_elements` and `tensor.extract`, which would support scf.forall in [RewriteTensorPointer.cpp](https://github.com/triton-lang/triton/blob/main/lib/Dialect/Triton/Transforms/RewriteTensorPointer.cpp#L515). Of course, there could be many other potential use cases as well.

Reply:
> Does Triton have any plans to use tensor.extract in the future to get a specific value from a tensor?

IMO, we don't have a plan to support it in the short term

Reply:
at this point no we don't think we want to support tensor.extract

Reply:
Get it!

Then, for this PR, I need to remove all the `tensor.extract` related code and simply add a `mask == cond` check in the [CombineSelectMaskedLoadPattern](https://github.com/triton-lang/triton/blob/main/lib/Dialect/Triton/Transforms/Combine.cpp#L124) pattern, along with additional tests.

Or, should I just close this PR?